volatility3/framework/automagic/linux.py

```diff
@@ -199,3 +214,212 @@ class LinuxSymbolFinder(symbol_finder.SymbolFinder):
     @classmethod
     def find_aslr(cls, *args):
         return LinuxIntelStacker.find_aslr(*args)[1]
+
+
+class LinuxIntelVMCOREINFOStacker(interfaces.automagic.StackerLayerInterface):
+    stack_order = 34
+    exclusion_list = ["mac", "windows"]
+
+    @staticmethod
+    def _check_versions() -> bool:
+        """Verify the versions of the required modules"""
+        # Check VMCOREINFO API version
+        vmcoreinfo_version_required = (1, 0, 0)
+        if not requirements.VersionRequirement.matches_required(
+            vmcoreinfo_version_required, linux.VMCoreInfo.version
+        ):
+            vollog.info(
+                "VMCOREINFO version not suitable: required %s found %s",
+                vmcoreinfo_version_required,
+                linux.VMCoreInfo.version,
+            )
+            return False
+
+        return True
+
+    @classmethod
+    def stack(
+        cls,
+        context: interfaces.context.ContextInterface,
+        layer_name: str,
+        progress_callback: constants.ProgressCallback = None,
+    ) -> Optional[interfaces.layers.DataLayerInterface]:
+        """Attempts to identify linux within this layer."""
+
+        # Verify the versions of the required modules
+        if not cls._check_versions():
+            return None
+
+        # Bail out by default unless we can stack properly
+        layer = context.layers[layer_name]
+
+        # Never stack on top of an intel layer
+        # FIXME: Find a way to improve this check
+        if isinstance(layer, intel.Intel):
```

Comment:
Yep, this *will* go wrong for virtualized hosts.  We're still a little way off supporting them yet, but definitely one not to forget about...